#20 메뉴 삭제

taekyun95 opened the pull request:
개요
메뉴 삭제

이슈 티켓
#20 메뉴 삭제 구현

작업 내용
- 메뉴 삭제 기능 추가
- 메뉴 유효성 추가
    - 메뉴 id가 존하는지 체크
    - 메뉴 id와 가게 id가 존재하는지 체크

taekyun95 committed:
- 메뉴 삭제 기능 추가
- 메뉴 유효성 추가
    - 메뉴 id가 존하는지 체크
    - 메뉴 id와 가게 id가 존재하는지 체크

taekyun95 committed:
#20 잘못된 테스트 수정

bgpark82 commented on src/main/java/com/flab/commerce/config/SecurityConfig.java:
```diff
@@ -25,6 +26,7 @@ protected void configure(HttpSecurity http) throws Exception {
         .antMatchers(HttpMethod.GET, "/users/logout").permitAll()
         .antMatchers(HttpMethod.POST, "/owners", "/owners/login").permitAll()
         .antMatchers(HttpMethod.GET, "/owners/logout").permitAll()
+        .antMatchers(HttpMethod.DELETE, "/stores/*/menus/*").hasAuthority(Constants.ROLE_OWNER)
```
Controller에서 @Secured라는 어노테이션으로 접근 권한을 관리하는 것도 좋을 것 같습니다

https://docs.spring.io/spring-security/site/docs/3.2.8.RELEASE/apidocs/org/springframework/security/access/annotation/Secured.html

taekyun95 replied:
`securityConfig`에 모아 관리하는게 좋지않을까요?

bgpark82 replied:
지금은 복잡하지 않아서 그렇게 해도 좋을 것 같습니다
단지 url이 복잡해지고 그에 따라 Role이 달라지면 그때마다 설정값에 추가해줘야 되기 때문에 그때는 @Secured를 사용해도 좋을 것 같습니다

f-lab-jason commented on src/main/java/com/flab/commerce/domain/store/StoreService.java:
```diff
@@ -1,16 +1,32 @@
+  public void validateOwnerStore(Long ownerId, Long storeId) {
+    validateStore(storeId);
+
+    if (!storeMapper.idAndOwnerIdExists(storeId, ownerId)) {
```
이친구는 어디에 있는건가요?

taekyun95 committed:
#21 메뉴 패치 (#26)

* pick 5dc1c46 #20 메뉴 삭제
    pick 064a2b0 #21 메뉴 업데이트

* * #27 옵션그룹등록

* #27 옵션그룹 목록 조회

- 조건은 가게 ID

* #27 옵션그룹 삭제

- 옵션그룹이 존하는지 검증하는 기능 추가
- 다른 가게의 옵션그룹인지 검증하는 기능 추가

* #27 옵션그룹 업데이트
